test/test_multirate.py

```diff
@@ -355,6 +355,7 @@ def am_solver_sr_hook(solve_expr, solve_var, solver_id, guess):
 
 
 def test_implicit_single_rate_identical(order=3, hist_length=3):
+    pytest.importorskip("scipy")
```

Comment:
Not a fan. Just add scipy [here](https://github.com/inducer/leap/blob/f1b5a3efc3ef2b50e2fc489523fadcab3ced07d1/.github/workflows/ci.yml#L56).